delphi-checks/src/main/java/au/com/integradev/delphi/checks/RedundantInheritedCheck.java

```diff
@@ -0,0 +1,144 @@
+import au.com.integradev.delphi.utils.RoutineUtils;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.communitydelphi.api.ast.CompoundStatementNode;
+import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
+import org.sonar.plugins.communitydelphi.api.ast.ExpressionStatementNode;
+import org.sonar.plugins.communitydelphi.api.ast.RoutineImplementationNode;
+import org.sonar.plugins.communitydelphi.api.ast.StatementListNode;
+import org.sonar.plugins.communitydelphi.api.ast.StatementNode;
+import org.sonar.plugins.communitydelphi.api.ast.utils.ExpressionNodeUtils;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
+import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
+import org.sonar.plugins.communitydelphi.api.check.FilePosition;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFix;
+import org.sonar.plugins.communitydelphi.api.reporting.QuickFixEdit;
+import org.sonar.plugins.communitydelphi.api.symbol.declaration.RoutineNameDeclaration;
+import org.sonar.plugins.communitydelphi.api.token.DelphiTokenType;
+
+@Rule(key = "RedundantInherited")
+public class RedundantInheritedCheck extends DelphiCheck {
+  private static final String MESSAGE = "Remove this redundant inherited call.";
+
+  @Override
+  public DelphiCheckContext visit(RoutineImplementationNode routine, DelphiCheckContext context) {
+    for (DelphiNode violationNode : findViolations(routine)) {
+      context
+          .newIssue()
+          .onNode(violationNode)
+          .withMessage(MESSAGE)
+          .withQuickFixes(getQuickFixes(violationNode))
+          .report();
+    }
+    return super.visit(routine, context);
+  }
+
+  private static List<QuickFix> getQuickFixes(DelphiNode violationNode) {
+    DelphiNode statement = violationNode;
+    DelphiNode parent = violationNode.getParent();
+    while (!(parent instanceof StatementListNode)) {
+      statement = parent;
+      parent = parent.getParent();
+      if (parent == null) {
+        return null;
+      }
+    }
+    StatementListNode statementListNode = (StatementListNode) parent;
+    int statementIndex = statementListNode.getStatements().indexOf(statement);
+
+    int startLine = statement.getBeginLine();
+    int startCol = statement.getBeginColumn();
+    int endLine = statement.getEndLine();
+    int endCol = statement.getEndColumn();
+
+    if (statementListNode.getStatements().size() > statementIndex + 1) {
+      StatementNode statementNode = statementListNode.getStatements().get(statementIndex + 1);
+      endLine = statementNode.getBeginLine();
+      endCol = statementNode.getBeginColumn();
+    } else {
+      if (statementIndex > 0) {
+        StatementNode previousStatement = statementListNode.getStatements().get(statementIndex - 1);
+        FilePosition startPosition = skipTrailingSemicolon(statementListNode, previousStatement);
```

Comment:
I think skipping the trailing semicolon in all these cases is probably unnecessary computation. 

For a statement `x`, you should be able to:

1. If there is a statement `x + 1`, remove from the start of statement node `x` to the start of statement node `x + 1`
2. If there is a statement `x - 1`, remove from the end of statement node `x - 1` to the end of statement node `x`
3.  Otherwise, remove statement node `x` and any trailing semicolons

This would look like:

```delphi
// before quick fix
destructor TFoo.Destroy;
begin
  FreeAndNil(Bar);
  inherited;
```
```delphi
// during quick fix (artist's impression)
destructor TFoo.Destroy;
begin
  FreeAndNil(Bar)[;
  inherited];
```
```delphi
// after quick fix
destructor TFoo.Destroy;
begin
  FreeAndNil(Bar);
end;
```

It is true that this removes comments and compiler directives - as do all quick fixes. I think we're not currently equipped to make changes that delicately, nor do I think we necessarily need to.